source/loader/layers/sanitizer/asan_interceptor.hpp

```diff
@@ -106,17 +108,15 @@ struct ContextInfo {
     std::vector<ur_device_handle_t> DeviceList;
     std::unordered_map<ur_device_handle_t, AllocInfoList> AllocInfosMap;
 
+    AsanStats Stats;
+
     explicit ContextInfo(ur_context_handle_t Context) : Handle(Context) {
         [[maybe_unused]] auto Result =
             getContext()->urDdiTable.Context.pfnRetain(Context);
         assert(Result == UR_RESULT_SUCCESS);
     }
 
-    ~ContextInfo() {
-        [[maybe_unused]] auto Result =
-            getContext()->urDdiTable.Context.pfnRelease(Handle);
-        assert(Result == UR_RESULT_SUCCESS);
-    }
+    ~ContextInfo();
```

Comment:
Why do we retain context but do not release them?

Reply:
It released here:

https://github.com/oneapi-src/unified-runtime/pull/1869/files#diff-cdb245c314b0deaa1e8277e335e881a8f7985456bb01ebcfe92aa4321b1de6ceR184-R186